client/cli/src/lib.rs

```diff
@@ -870,14 +876,17 @@ impl FedimintCli {
                     new_gateway: (gateway_json),
                 })
             }
-            Command::Backup => cli
-                .build_client(&self.module_gens)
-                .await?
-                .mint_client()
-                .back_up_ecash_to_federation()
-                .await
-                .map(|_| CliOutput::Backup)
-                .map_err_cli_msg(CliErrorKind::GeneralFederationError, "failed"),
+            Command::Backup { metadata } => {
+                let metadata = metadata_from_clap_cli(metadata)?;
+
+                cli.build_client(&self.module_gens)
+                    .await?
+                    .mint_client()
+                    .back_up_ecash_to_federation(Metadata::json_from(metadata))
+                    .await
+                    .map(|_| CliOutput::Backup)
+                    .map_err_cli_msg(CliErrorKind::GeneralFederationError, "failed")
```

Comment:
Do we know what exactly failed here (upload/network/dunno?)? Otherwise the exact error message is contained in the error itself, right?

Reply:
The error does have the `raw_error` so the original error will be visible.

```
    fn map_err_cli_msg(self, kind: CliErrorKind, msg: impl Into<String>) -> Result<O, CliError> {
        self.map_err(|e| CliError {
            kind,
            message: msg.into(),
            raw_error: Some(e.into()),
        })
    }
```

Reply:
:+1: `"failed"` as message in an error seems a bit useless, though

since we call `.back_up_ecash_to_federation(Metadata::json_from(metadata))` perhaps `ecash backup failed`?